### PR TITLE
Ddl plouie.dom 39750.no queue declare on every publish

### PR DESCRIPTION
**Pull request by ddl-plouie**
**What this PR does / why we need it**:
Removed QueueDeclare() on every message.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: executor/api/grpc/kfserving/inference/grpc_service.pb.go [hunk 1]
```diff
@@ -24,11 +24,11 @@ var _ = math.Inf
 // proto package needs to be updated.
 const _ = proto.ProtoPackageIsVersion3 // please upgrade the proto package
 
-//@@
```
Comment (dmcwhorter-ddl) [inline executor/api/grpc/kfserving/inference/grpc_service.pb.go]: please revert changes to this file

File: executor/api/grpc/kfserving/inference/model_config.pb.go [hunk 2]
```diff
@@ -20,11 +20,11 @@ var _ = math.Inf
 // proto package needs to be updated.
 const _ = proto.ProtoPackageIsVersion3 // please upgrade the proto package
 
-//@@
```
Comment (dmcwhorter-ddl) [inline executor/api/grpc/kfserving/inference/model_config.pb.go]: please revert changes to this file

File: executor/api/rabbitmq/publisher_test.go [hunk 3]
```diff
@@ -22,9 +22,9 @@ func TestPublisher(t *testing.T) {
 	t.Run("success", func(t *testing.T) {
 		mockChan := &mockChannel{}
 
-		mockChan.On("QueueDeclare", queueName, true, false, false, false,
-			queueArgs).Return(amqp.Queue{Name: queueName},
-			nil)
+		//mockChan.On("QueueDeclare", queueName, true, false, false, false,
```
Comment (dmcwhorter-ddl) [inline executor/api/rabbitmq/publisher_test.go]: please remove commented code

File: executor/api/rabbitmq/publisher_test.go [hunk 4]
```diff
@@ -71,30 +67,11 @@ func TestPublisher(t *testing.T) {
...
-
-		mockChan.AssertExpectations(t)
-	})
-
 	t.Run("connection_closed", func(t *testing.T) {
 		f, reset := setupConnect(func(adapter *mockDialerAdapter, conn *mockConnection, channel *mockChannel) {
-			channel.On("QueueDeclare", queueName, true, false, false, false,
-				queueArgs).Return(amqp.Queue{Name: queueName},
-				nil)
+			//channel.On("QueueDeclare", queueName, true, false, false, false,
```
Comment (dmcwhorter-ddl) [inline executor/api/rabbitmq/publisher_test.go]: please remove commented code

File: executor/api/rabbitmq/publisher_test.go [hunk 5]
```diff
@@ -71,30 +67,11 @@ func TestPublisher(t *testing.T) {
...
+			//	queueArgs).Return(amqp.Queue{Name: queueName},
```
Comment (dmcwhorter-ddl) [inline executor/api/rabbitmq/publisher_test.go]: same

File: executor/api/rabbitmq/server_test.go [hunk 6]
```diff
@@ -87,16 +86,6 @@ func TestRabbitMqServer(t *testing.T) {
 	testPuid := guuid.New().String()
 	testHeaders := map[string]interface{}{payload.SeldonPUIDHeader: testPuid}
 
-	invalidErrorJsonResponse := fmt.Sprintf(
```
Comment (dmcwhorter-ddl) [inline executor/api/rabbitmq/server_test.go]: this should still be there, this is unrelated to QueueDeclare

File: executor/api/rabbitmq/server_test.go [hunk 7]
```diff
@@ -129,89 +118,6 @@ func TestRabbitMqServer(t *testing.T) {
 		assert.Equal(t, fullHealthCheck, server.FullHealthCheck)
 	})
 
-	/*
-	 * This makes sure the Serve() and predictAndPublishResponse() code runs and makes the proper calls
-	 * by hacking a bunch of mocks.
-	 * It is not doing anything to validate the messages are properly processed.  That's challenging in a
-	 * unit test since the code connects to RabbitMQ.
-	 */
-	t.Run("serve", func(t *testing.T) {
```
Comment (dmcwhorter-ddl) [inline executor/api/rabbitmq/server_test.go]: This is one of the main tests for the server, it should be updated to work with the changes and not removed

File: executor/api/rabbitmq/server_test.go [hunk 8]
```diff
@@ -129,89 +118,6 @@ func TestRabbitMqServer(t *testing.T) {
...
-		assert.NoError(t, err)
-
-		mockRmqChan.AssertExpectations(t)
-	})
-
-	/*
-	 * This makes sure the Serve(), predictAndPublishResponse(), and createAndPublishErrorResponse() code runs and
-	 * makes the proper calls and returns an appropriate error message by hacking a bunch of mocks.
-	 */
-	t.Run("serveError", func(t *testing.T) {
```
Comment (dmcwhorter-ddl) [inline executor/api/rabbitmq/server_test.go]: this should be updated and not removed

**Review by dmcwhorter-ddl — CHANGES_REQUESTED** [review]
Please fix test cases

**Comment by dmcwhorter-ddl** [comment]
@ddl-plouie the integration test is failing (circleCI workflow above)

File: executor/api/grpc/kfserving/inference/grpc_service.pb.go (hunk 1 above)
Comment (ddl-plouie) [reply executor/api/grpc/kfserving/inference/grpc_service.pb.go]: Discussed

File: executor/api/rabbitmq/publisher_test.go (hunk 3 above)
Comment (ddl-plouie) [reply executor/api/rabbitmq/publisher_test.go]: Discussed

File: executor/api/rabbitmq/publisher_test.go (hunk 5 above)
Comment (ddl-plouie) [reply executor/api/rabbitmq/publisher_test.go]: Done

File: executor/api/rabbitmq/server_test.go (hunk 7 above)
Comment (ddl-plouie) [reply executor/api/rabbitmq/server_test.go]: Done

File: executor/api/rabbitmq/server_test.go (hunk 8 above)
Comment (ddl-plouie) [reply executor/api/rabbitmq/server_test.go]: Done

**Review by dmcwhorter-ddl — APPROVED** [review]

**Pull request merged**